power_grid_model_c/power_grid_model/include/power_grid_model/common/common.hpp

```diff
@@ -30,6 +30,14 @@ struct Idx2D {
     friend constexpr bool operator==(Idx2D x, Idx2D y) = default;
 };
 
+struct Idx2DHash {
+    std::size_t operator()(const Idx2D& idx) const {
```

Comment:
i think this can be `constexpr`

Reply:
If only `std::hash<Idx>` is `constexpr`

Reply:
welp, guess we'll have to wait until C++29 then... https://www.reddit.com/r/cpp_questions/comments/ld7yux/why_isnt_stdhash_constexpr/